### PR TITLE
 Validate birthDate and handle server error.

### PR DESCRIPTION
**Pull request by cdevent**
## Related Issue or Background Info

https://github.com/CDCgov/prime-simplereport/issues/442
https://github.com/CDCgov/prime-simplereport/issues/1398

## Changes Proposed

- Add validation for the birthDate field(min/max birthDate).
- Show toaster message on network error.
- redirect the user to the main page only if we receive 401 status 

## Screenshots / Demos
Screenshot 2021-05-07 at 11 52 26

### UI
- [ ] Any changes to the UI/UX have been approved by design 
- [ ] Any new or edited error messages have been approved by design 

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR
- [ ] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support

### CODE REVIEW

File: frontend/src/index.tsx [hunk 1]
```diff
@@ -56,11 +56,19 @@ const authMiddleware = new ApolloLink((operation, forward) => {
...
   if (networkError && process.env.REACT_APP_BASE_URL) {
-    console.error("network error", networkError);
-    console.log("redirecting to", process.env.REACT_APP_BASE_URL);
-    window.location.replace(process.env.REACT_APP_BASE_URL);
+    if ("statusCode" in networkError && networkError.statusCode === 401) {
+      console.warn("[UNATHORIZED_ACCESS] !!");
+      console.warn("redirect-to:", process.env.REACT_APP_BASE_URL);
+      window.location.replace(process.env.REACT_APP_BASE_URL);
+    } else {
+      showError(
```
Comment (timothybest-usds) [inline frontend/src/index.tsx:66]: Can we add a log statement here as well
```suggestion
     appInsights.trackException({error: networkError});
      showError(
```

**Review by timothybest-usds — CHANGES_REQUESTED** [review]

File: frontend/src/app/patients/personSchema.ts [hunk 2]
```diff
@@ -123,7 +123,7 @@ export const personSchema: yup.SchemaOf<RequiredPersonFields> = yup.object({
   firstName: yup.string().required(),
   middleName: yup.string().nullable(),
   lastName: yup.string().required(),
-  birthDate: yup.string().required(),
+  birthDate: yup.date().required().max(new Date()).min(new Date("1/1/1900")),
```
Comment (nickscialli-usds) [inline frontend/src/app/patients/personSchema.ts:126]: The validation text (specified lower in this file) doesn't make sense any more "Date of birth is missing or incorrectly formatted"

I recommend changing it to "Date of birth must be correctly formatted (MM/DD/YYY) and in the past"

Comment (cdevent) [reply frontend/src/app/patients/personSchema.ts:126]: Yeah, make sense. But in that case, we do not inform that this is a mandatory field. only red asterisk near field.

Comment (nickscialli-usds) [reply frontend/src/app/patients/personSchema.ts:126]: True! Maybe something like "Date of birth must be present, correctly formatted (MM/DD/YYY), and in the past"

**Review by timothybest-usds — APPROVED** [review]

**Review by nickscialli-usds — APPROVED** [review]

**Review by nullflux — APPROVED** [review]
In the future it'd be helpful if you'd try to separate functional changesets into separate PR's – but otherwise this LGTM

File: frontend/src/app/patients/personSchema.ts (hunk 2 above)
Comment (maya) [reply frontend/src/app/patients/personSchema.ts:128]: (Just seeing this now b/c of the incident report), but regarding the error copy, "Date of birth must be present," is redundant b/c an error implicitly states this, so I'd remove that part.

Note: we have a checklist item for design to review new content too, /cc @Rebecca-LM

Comment (maya) [reply frontend/src/app/patients/personSchema.ts:128]: Even better would be an error message for when it's not present:
"Enter your date of birth"

And one when it's incorrectly formated: 
"Enter your date of birth in the correct format (MM/DD/YYYY)"

When they made both mistakes:
"Enter your date of birth in the correct format (MM/DD/YYYY) and in the past"

etc... you want the error to reflect whatever they did wrong so it's helpful.

**Pull request merged**